src_c/time.c

```diff
@@ -513,18 +553,27 @@ MODINIT_DEFINE(time)
     if (PyErr_Occurred()) {
         MODINIT_ERROR;
     }
-#if IS_SDLv2
+
     import_pygame_event();
     if (PyErr_Occurred()) {
         MODINIT_ERROR;
     }
-#endif /* IS_SDLv2 */
 
     /* type preparation */
     if (PyType_Ready(&PyClock_Type) < 0) {
         MODINIT_ERROR;
     }
 
+    /* register cleanup function for event timer holding structure,
+     * allocate a mutex for this structure too */
+    if (!timermutex && !pg_event_timer) {
+        timermutex = SDL_CreateMutex();
+        if (!timermutex) {
```

Comment:
Is this correct? If the timermutex fails, we do anyway `pg_RegisterQuit(_pg_event_timer_cleanup);`?

Reply:
If timermutex fails, then modinit_error will exit the function, right?

Reply:
Indeed, it will.

Probably need to set the python error here with the SDL error?
```c
PyErr_SetString(PyExc_ImportError, SDL_GetError());
```

Reply:
Ah I missed that. Will add

Reply:
Added :)